packages/flutter_migrate/lib/src/commands/start.dart

```diff
@@ -0,0 +1,340 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:process/process.dart';
+
+import '../base/command.dart';
+import '../base/file_system.dart';
+import '../base/logger.dart';
+import '../base/project.dart';
+
+import '../compute.dart';
+import '../environment.dart';
+import '../manifest.dart';
+import '../result.dart';
+import '../utils.dart';
+
+class MigrateStartCommand extends MigrateCommand {
+  MigrateStartCommand({
+    bool verbose = false,
+    required this.logger,
+    required this.fileSystem,
+    required this.processManager,
+  })  : _verbose = verbose,
+        migrateUtils = MigrateUtils(
+          logger: logger,
+          fileSystem: fileSystem,
+          processManager: processManager,
+        ) {
+    argParser.addOption(
+      'staging-directory',
+      help:
+          'Specifies the custom migration staging directory used to stage and edit proposed changes. '
+          'This path can be absolute or relative to the flutter project root.',
+      valueHelp: 'path',
+    );
+    argParser.addOption(
+      'project-directory',
+      help: 'The root directory of the flutter project.',
+      valueHelp: 'path',
+    );
+    argParser.addOption(
+      'platforms',
+      help:
+          'Restrict the tool to only migrating the listed platforms. By default all platforms generated by '
+          'flutter create will be migrated. To indicate the project root, use the `root` platform',
```

Comment:
what does it mean to migrate the project root?

Reply:
This just refers to the files under the project root dir that are not part of a platform folder. I can clarify in the doc.